app/webpacker/components/RegistrationsV2/Register/RegistrationMessage.jsx

```diff
@@ -3,53 +3,47 @@ import React, { useEffect } from 'react';
 import { useDispatch, useStore } from '../../../lib/providers/StoreProvider';
 import I18n from '../../../lib/i18n';
 
-export const setMessage = (key, type, params) => ({
-  payload: {
-    key,
-    type,
-    params,
-  },
+/** To show multiple messages, use `showMessages` instead. */
+export const showMessage = (key, type, params) => ({
+  newMessages: [{ key, type, params }],
 });
 
-export const clearMessage = () => ({
-  payload: {
-    message: null,
-  },
-});
+export const showMessages = (messages) => ({ newMessages: messages });
+
+export const clearAllMessages = () => ({ messages: [] });
+
+const clearMessage = (id) => ({ toClear: [id] });
+
+const clearMessages = (ids) => ({ toClear: ids });
 
 export default function RegistrationMessage() {
-  const { message } = useStore();
+  const { messages } = useStore();
   const dispatch = useDispatch();
 
+  const positiveMessages = messages.filter(({ type }) => type === 'positive');
+
   useEffect(() => {
-    // Don't clear negative Messages automatically
-    if (message?.key && message.type !== 'negative') {
+    if (positiveMessages.length > 0) {
       setTimeout(() => {
-        dispatch({ payload: { message: null } });
+        // some may already be cleared by an earlier timeout; that's fine
+        dispatch(clearMessages(positiveMessages.map(({ id }) => id)));
       }, 4000);
     }
-  }, [dispatch, message]);
-
-  if (!message?.key) return null;
-
-  if (Array.isArray(message.key)) {
-    return message.key.map((key) => (
-      <Message
-        positive={message.type === 'positive'}
-        negative={message.type === 'negative'}
-      >
-        {I18n.t(key, message.params)}
-      </Message>
-    ));
-  }
-
-  return (
+  }, [dispatch, positiveMessages]);
+
+  if (messages.length === 0) return null;
+
+  return messages.map(({
+    id, key, type, params,
+  }) => (
     <Message
-      style={{ margin: 0 }}
-      positive={message.type === 'positive'}
-      negative={message.type === 'negative'}
+      key={id}
+      style={{ margin: messages.length === 1 ? 0 : undefined }}
+      positive={type === 'positive'}
+      negative={type === 'negative'}
+      onDismiss={() => dispatch(clearMessage(id))}
```

Comment:
Does passing an `onDismiss` automatically give the message a small "cross" icon to clear it?
And would it make sense to make positive messages non-dismissable since they clear after 4 seconds anyways?

Reply:
Yes.
I thought about it originally. I've done it now.